plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -266,16 +273,25 @@ private void putAssetPayload(final MavenPath path,
         BlobStore.CREATED_BY_HEADER, "unknown"
     );
 
-    try (InputStream inputStream = content.openInputStream()) {
+    try (InputStream inputStream = payload.openInputStream()) {
       try (TempStreamSupplier supplier = new TempStreamSupplier(inputStream)) {
-        final String contentType = determineContentType(path, supplier, content.getContentType());
+        final String contentType = determineContentType(path, supplier, payload.getContentType());
         try (InputStream is = supplier.get()) {
           tx.setBlob(is, headers, asset, HASH_ALGORITHMS, contentType);
         }
       }
     }
 
-    asset.formatAttributes().set(P_CONTENT_LAST_MODIFIED, new Date());
+    final NestedAttributesMap formatAttributes = asset.formatAttributes();
+    if (payload instanceof Content) {
```

Comment:
No, method work with both, Payload and Content. It is just matter of what arrives: in case of proxy, we may receive "extra information" that storage must save for later reuse, like the content last-modified and content etag (opaque identifier that might be sent by remote). The actual payload was already stored in a blob, and blob got attached to asset above, and P_LAST_UPDATED is automatically maintained on asset.

Again, these two attributes are optional, that might or might not be sent by upstream remote server.